### PR TITLE
DCOS-21044: Fix SDK tasks health indication

### PR DESCRIPTION
**Pull request by nLight**
⚠️ Back port to 1.11 needed
*TL;DR*: all tasks launched by an SDK framework should be shown as healthy

We don't get info whether a task launched by an SDK framework is healthy or not, but it's confusing for our customers to see gray dots. So when we detect a task is an SDK task and it is running we say it's healthy. 

In our performance PR I introduced the regression by moving the flagging logic and eventually breaking it.

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [x] If this is a regression, did you write a test to catch this in the future?

Closes DCOS-21044

### CODE REVIEW

**Comment by DanielMSchmidt** [comment]
Run integration tests

**Comment by DanielMSchmidt** [comment]
run integration tests

**Comment by DanielMSchmidt** [comment]
Run integration tests

**Comment by DanielMSchmidt** [comment]
run integration tests

**Comment by juliangieseke** [comment]
run integration tests

**Comment by juliangieseke** [comment]
run all tests

**Comment by juliangieseke** [comment]
@nLight looks like this particular PR cant trigger Jenkins anymore, do you have an idea what happened here?

**Comment by Fabs** [comment]
The failing test looks flake.

Run integration tests

**Comment by Fabs** [comment]
run integration tests

**Comment by Fabs** [comment]
Closed because we were not able to run integration tests on this one anymore.
